src/dbc.rs

```diff
@@ -66,9 +66,15 @@ use crate::{
 #[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
 #[derive(Debug, Clone, Eq, PartialEq)]
 pub struct Dbc {
+    /// Encrypted information for and about this DBC's owner
     pub content: DbcContent,
+    /// The public key (owner) of this DBC, used for SpentProofs
+    pub public_key: PublicKey,
+    /// The transaction where this DBC was created
     pub transaction: DbcTransaction,
+    /// The transaction's input's SpentProofs
     pub spent_proofs: BTreeSet<SpentProof>,
```

Comment:
I would even consider renaming these to:
`input_spent_proofs` and `input_spent_txs` for additional clarity.